src/main/java/org/broadinstitute/hellbender/tools/exome/IntervalBackedExonCollection.java

```diff
@@ -33,43 +35,65 @@
      */
     private int lastBinarySearchResult = -1;
 
+
+    /**
+     * Comparator use to sort and search intervals.
+     * <p>
+     * Since we are guaranteed not to have overlaps between intervals (checked at construction),
+     * we only need to pay attention to the contig (sorted alphanumerically) and the start position.
+     * </p>
+     */
+    private Comparator<SimpleInterval> INTERVAL_COMPARATOR = (a,b) -> {
+        if (a == null) {
+            throw new IllegalArgumentException("the left element cannot be null");
+        }
+        if (b == null ) {
+            throw new IllegalArgumentException("the right element cannot be null");
+        }
+        final int contigComparison = a.getContig().compareTo(b.getContig());
+        if (contigComparison == 0) {
+            return Integer.compare(a.getStart(),b.getStart());
+        } else {
+            return contigComparison;
+        }
+    };
+
     /**
      * Creates a exon data-base give a sorted list of intervals.
      *
+     * <p>
+     *     Intervals will be sorted by their contig id lexicographical order.
+     * </p>
+     *
      * @param intervals the input interval list. Is assumed to be sorted
      * @throws IllegalArgumentException if {@code intervals} is {@code null}.
      */
-    public IntervalBackedExonCollection(final List<GenomeLoc> intervals) {
+    public IntervalBackedExonCollection(final List<SimpleInterval> intervals) {
         if (intervals == null) {
             throw new IllegalArgumentException("the input intervals cannot be null");
         }
-        checkInputIntervalCoherence(intervals);
-        this.sortedIntervals = Collections.unmodifiableList(new ArrayList<>(intervals));
-        this.intervalsByName = composeIntervalsByName(intervals);
+        if (intervals.contains(null)) {
+            throw new IllegalArgumentException("the input cannot contain null");
```

Comment:
Leave the check if you think it's useful.  I think it's unlikely to come up often, but it would be confusing when it happened.